src/marlin_pc/data_structures.rs

```diff
@@ -126,6 +126,20 @@ impl<E: PairingEngine> PCVerifierKey for VerifierKey<E> {
     }
 }
 
+impl<E: PairingEngine> ToBytes for VerifierKey<E> {
+    fn write<W: algebra_core::io::Write>(&self, mut writer: W) -> algebra_core::io::Result<()> {
+        self.vk.write(&mut writer)?;
+        if let Some(degree_bounds_and_shift_powers) = &self.degree_bounds_and_shift_powers {
```

Comment:
You might want to additionally include the length during serialization here

Reply:
Added!

Reply:
I'm curious. What is the advantage of including the length here? Isn't it possible to derive if we already know the byte length of all of the other members?